packages/transport/src/modules/accounts/IdentityDeletionProcessController.ts

```diff
@@ -0,0 +1,90 @@
+import { IdentityDeletionProcessClient } from "./backbone/IdentityDeletionProcessClient";
+import { IdentityDeletionProcess } from "./data/IdentityDeletionProcess";
+import { IdentityDeletionProcessStatus } from "./data/IdentityDeletionProcessStatus";
+
+export class IdentityDeletionProcessController extends TransportController {
+    public identityDeletionProcessClient: IdentityDeletionProcessClient;
+    public identityDeletionProcessCollection: SynchronizedCollection;
+
+    public constructor(parent: AccountController) {
+        super(ControllerName.Identity, parent);
+
+        this.identityDeletionProcessClient = new IdentityDeletionProcessClient(this.config, this.parent.authenticator);
+    }
+
+    @log()
+    public override async init(): Promise<IdentityDeletionProcessController> {
+        await super.init();
+
+        this.identityDeletionProcessCollection = await this.parent.getSynchronizedCollection("IdentityDeletionProcesses");
+        return this;
+    }
+
+    public async getIdentityDeletionProcess(identityDeletionProcessId: string): Promise<IdentityDeletionProcess | undefined> {
+        const identityDeletionProcess = await this.identityDeletionProcessCollection.findOne({ id: identityDeletionProcessId });
+        return identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined;
+    }
+
+    public async getIdentityDeletionProcesses(): Promise<IdentityDeletionProcess[]> {
+        return (await this.identityDeletionProcessCollection.find())
+            .map((identityDeletionProcess) => (identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined))
+            .filter((identityDeletionProcess) => !!identityDeletionProcess) as IdentityDeletionProcess[];
+    }
+
+    public async getApprovedIdentityDeletionProcess(): Promise<IdentityDeletionProcess | undefined> {
+        const identityDeletionProcess = await this.identityDeletionProcessCollection.findOne({
+            status: IdentityDeletionProcessStatus.Approved
+        });
+        return identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined;
+    }
+
+    public async getWaitingIdentityDeletionProcess(): Promise<IdentityDeletionProcess | undefined> {
+        const identityDeletionProcess = await this.identityDeletionProcessCollection.findOne({
+            status: IdentityDeletionProcessStatus.WaitingForApproval
+        });
+
+        return identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined;
+    }
+
+    private async updateIdentityDeletionProcess(identityDeletionProcess: IdentityDeletionProcess): Promise<void> {
+        const oldIdentityDeletion = await this.identityDeletionProcessCollection.findOne({ id: identityDeletionProcess.id.toString() });
+        await this.identityDeletionProcessCollection.update(oldIdentityDeletion, identityDeletionProcess);
```

Comment:
Please be VERY cautious with the SynchronizedCollection. You have to work _**ANYTIME**_ with CoreSynchronizable Objects - typescript doesn't detect this error, but if you would have tested this with synchronization enabled that should have blown up.

Reply:
(to fix this pls call getIdentityDeletionProcess in the line before)

and make sure that the process is defined.